src/main/java/com/gmail/goosius/siegewar/SiegeController.java

```diff
@@ -258,6 +259,8 @@ public static void removeSiege(Siege siege) {
 				e.printStackTrace();
 			}
 		}
+		//Call event
+		Bukkit.getPluginManager().callEvent(new SiegeRemoveEvent(siege));

```

Comment:
The way this was being thrown before had it happening after the town.save() and the event says it is thrown after removal.

Probably worth moving this to below the town.save();